manager/main.go

```diff
@@ -142,6 +142,11 @@ func main() {
 	setupChecks(setupLog, mgr)
 	//+kubebuilder:scaffold:builder
 
+	if err := mgr.GetFieldIndexer().
```

Comment:
Can we leave a comment here about why we want this index and what is being used for?
It might not be obvious at a first glance.